sdk/servicebus/azure-servicebus/api_review.py

```diff
@@ -0,0 +1,588 @@
+)
+'''
+
+with queue_sender:
+    # 1.1 send single
+    message = Message("Test message")
+    queue_sender.send(message)
+
+    # 1.2 send list
+    messages = [Message("Test message {}".format(i)) for i in range(10)]
+    queue_sender.send(messages)
+
+    # 1.3 schedule
+    schedule_message = Message("scheduled_message")
+    enqueue_time = datetime.utcnow() + timedelta(minutes=10)
+    sequence_number = queue_sender.schedule(schedule_message, enqueue_time)
+
+    # 1.4 cancel schedule
+    queue_sender.cancel_scheduled_messages(*sequence_number)
+
+# 2 Receive
+
+queue_receiver = ServiceBusReceiverClient.from_queue(
+    fully_qualified_namespace="fully_qualified_namespace",
+    queue_name="queue_name",
+    credential=ServiceBusSharedKeyCredential("policy", "key")
+)
+
+'''
+subscription_receiver = ServiceBusReceiverClient.from_queue(
+    fully_qualified_namespace="fully_qualified_namespace",
+    topic_name="topic_name",
+    subscription_name="subscription_name"
+    credential=ServiceBusSharedKeyCredential("policy", "key")
+)
+'''
+
+# 2.1 peek
+with queue_receiver:
+    msgs = queue_receiver.peek(count=10)
+
+# 2.2 iterator receive
+with queue_receiver:
+    for msg in queue_receiver:
+        print(message)
+        msg.complete()
+        # msg.renew_lock()
+        # msg.abondon()
+        # msg.defer()
+        # msg.dead_letter()
+
+# 2.3 pull mode receive
+with queue_receiver:
+    msgs = queue_receiver.receive(max_batch_size=10)
+    for msg in msgs:
+        msg.complete()
+
+# 2.4 receive deferred letter
+with queue_receiver:
+    defered_sequence_numbers = [xxxx]
+    deferred = queue_receiver.receive_deferred_messages(defered_sequence_numbers)
+    queue_receiver.settle_deferred_messages('completed', deferred)
+
+
+### Session operation
+
+
+# 1 Send
+topic_sender = SeviceBusSenderClient.from_topic
+(
+    fully_qualified_namespace="fully_qualified_namepsace",
+    topic_name="topic_name",
+    credential=ServiceBusSharedKeyCredential("policy", "key")
+)
+
+with topic_sender:
+    # 1.1 send single
+    message = Message("Test message")
+    # message.session_id = "test_session"
+    topic_sender.send(message, session="test_session")
+
+    # 1.2 send list
+    messages = [Message("Test message {}".format(i)) for i in range(10)]
+    topic_sender.send(messages, session="test_session")
+
+    # 1.3 schedule
+    schedule_message = Message("scheduled_message")
+    enqueue_time = datetime.utcnow() + timedelta(minutes=10)
+    sequence_number = topic_sender.schedule(schedule_message, enqueue_time, session="test_session")
+
+    # 1.4 cancel schedule
+    topic_sender.cancel_scheduled_messages(*sequence_number)
+
+# 2 Receive
+subscription_receiver = ServiceBusReceiverClient.from_topic_subscription
+(
+    fully_qualified_namespace="fully_qualified_namespace",
+    topic_name="topic_name",
+    subscription_name="subscription_name",
+    credential=ServiceBusSharedKeyCredential("policy", "key"),
+    session="test_session"
```

Comment:
So we're keeping session just at the client level now?
If so - are there advantages to passing around a Session object?

Reply:
Yes, from my point this implies the concept of SessionReceiver when `session` is passed into the constructor. I like this because it clarifies Sessionful vs non-sessionful entity from the top level.  @KieranBrantnerMagee mentioned that keeping session at client level gives us the flexibility to move `session` into `receive` method in the future.

For the proposal -- pass around a Session object, are you saying that users would create a Session object and then pass into constructor? TBH I don't think it's a good idea.
Session objects would have service-request method, and these methods can't work until we establish the connection which is transparent to user.

Moreover, session-id (a string) is all we need to create a `SessionReceiver`, we don't have to bring extra burden (find the class Session, create a Session object) to a user to create a `SessionReceiver`.